extension/js/common/api/email-provider/gmail/google-auth.ts

```diff
@@ -181,33 +181,32 @@ export class GoogleAuth {
     return await GoogleAuth.newAuthPopup({ acctEmail, scopes: GoogleAuth.defaultScopes('openid'), save: false });
   };
 
-  private static getParameterByName = (search: string, name: string) => {
-    const match = RegExp('[?&]' + name + '=([^&]*)').exec(search);
-    return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
-  };
-
-  private static getAuthRes = async ({ acctEmail, save, authWindowResult }:
-    { acctEmail?: string, save: boolean, authWindowResult: Bm.AuthWindowResult }): Promise<AuthRes> => {
+  private static getAuthRes = async ({ acctEmail, save, state, authWindowResult }:
+    { acctEmail?: string, save: boolean, state: string, authWindowResult: Bm.AuthWindowResult }): Promise<AuthRes> => {
     try {
       if (!authWindowResult.url) {
         return { acctEmail, result: 'Denied', error: 'Invalid response url', id_token: undefined };
       }
       if (authWindowResult.error) {
         return { acctEmail, result: 'Denied', error: authWindowResult.error, id_token: undefined };
       }
-      const url = new URL(authWindowResult.url!);
-      const allowedScopes = this.getParameterByName(url.search, 'scope');
-      const code = this.getParameterByName(url.search, 'code') ?? '';
+      const uncheckedUrlParams = Url.parse(['scope', 'code', 'state'], authWindowResult.url!);
+      const allowedScopes = Assert.urlParamRequire.string(uncheckedUrlParams, 'scope');
+      const code = Assert.urlParamRequire.optionalString(uncheckedUrlParams, 'code');
+      const receivedState = Assert.urlParamRequire.string(uncheckedUrlParams, 'state');
```

Comment:
`Assert.urlParamRequire` throws error text that could be confusing for users, as it was meant for parsing URLs that we created ourselves. You'll notice this when reading the error text in `abortAndRenderErrOnUrlParamValMismatch`. It's not well documented.

But it's good to check it here. You could put these 4 lines in a try/catch, and if they throw error, return `        return { acctEmail, result: 'Error', error: `Could not parse URL returned from Google`, id_token: undefined };`